external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -270,17 +272,40 @@ public void setWaitingToEmit(ConsumerRecords<K,V> consumerRecords) {
 
     // ======== poll =========
     private ConsumerRecords<K, V> pollKafkaBroker() {
-        doSeekRetriableTopicPartitions();
-        if (refreshSubscriptionTimer.isExpiredResetOnTrue()) {
-            kafkaSpoutConfig.getSubscription().refreshAssignment();
+        Set<TopicPartition> retriableTopicPartitions = doSeekRetriableTopicPartitions();
+        Set<TopicPartition> partitionsToPause = getPartitionsToPauseToEnforceUncommittedOffsetsLimit(retriableTopicPartitions);
```

Comment:
let's replace lines 276 to 280 with the code I suggest in [here](https://github.com/apache/storm/pull/1924/files#r106557594)

!partitionsToPause.isEmpty() is not necessary